moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -736,21 +710,32 @@ void ServoCalcs::enforceSRDFAccelVelLimits(Eigen::ArrayXd& delta_theta)
       // Apply velocity bounds
       if (clip_velocity)
       {
-        // delta_theta = joint_velocity * delta_t
-        const double relative_change = (velocity_limit * parameters_.publish_period) / delta_theta(joint_delta_index);
-        // Avoid nan
-        if (fabs(relative_change) < 1)
-        {
-          delta_theta(joint_delta_index) = relative_change * delta_theta(joint_delta_index);
-          velocity(joint_delta_index) = relative_change * velocity(joint_delta_index);
-        }
+        const double relative_change =
+            fabs(velocity_limit * parameters_.publish_period) / fabs(delta_theta(joint_delta_index));
+
+        // Store the scaling factor if it's the smallest yet
+        if (relative_change < min_relative_velocity_all_joints)
+          min_relative_velocity_all_joints = relative_change;
       }
     }
     ++joint_delta_index;
   }
+
+  // Apply the velocity scaling to all joints
+  if (min_relative_velocity_all_joints < 1)
+  {
```

Comment:
Should there be a ROS_WARN or ROS_INFO here or would that be too spammy?

Reply:
too spammy IMO

Reply:
There are throttled versions of `ROS_WARN(..)` and `ROS_INFO(..)`, which could reduce the *spammy-ness*.

See `*_THROTTLE` et al.